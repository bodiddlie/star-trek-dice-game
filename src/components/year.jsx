import React from 'react';
import { connect } from 'react-redux';

export const Year = ({ year }) =>
  <div>
    Year - {year}
  </div>;

const mapStateToProps = state => ({
  year: state.year,
});

export default connect(mapStateToProps)(Year);
