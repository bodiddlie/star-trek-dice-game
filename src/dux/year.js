const initialState = 1;

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case ADVANCE_YEAR: {
      return state + 1;
    }
    default: {
      return state;
    }
  }
}

const ADVANCE_YEAR = '[Year] Advance Year';

export function advanceYear() {
  return { type: ADVANCE_YEAR };
}
