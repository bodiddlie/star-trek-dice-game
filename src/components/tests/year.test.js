import React from 'react';
import { shallow } from 'enzyme';

import { Year } from '../year';

describe('<Year />', () => {
  it('renders the current year track value', () => {
    const wrapper = shallow(<Year year={2} />);
    expect(wrapper).toMatchSnapshot();
  });
});
