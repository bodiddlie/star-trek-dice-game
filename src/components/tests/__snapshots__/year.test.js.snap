// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Year /> renders the current year track value 1`] = `
<div>
  Year - 
  2
</div>
`;
